source/exe/signal_action.cc

```diff
@@ -0,0 +1,119 @@
+#include "exe/signal_action.h"
+
+#include <signal.h>
+#include <sys/mman.h>
+
+#include "common/common/assert.h"
+
+constexpr int SignalAction::FATAL_SIGS[];
+
+void SignalAction::SigHandler(int sig, siginfo_t* info, void* context) {
+  void* error_pc = 0;
+
+  const ucontext_t* ucontext = reinterpret_cast<const ucontext_t*>(context);
+  if (ucontext != nullptr) {
+#ifdef REG_RIP
+    // x86_64
+    error_pc = reinterpret_cast<void*>(ucontext->uc_mcontext.gregs[REG_RIP]);
+#elif defined(REG_EIP)
```

Comment:
nit: Can we just drop all of the elif clauses and have a single warning without the commented out code?